src/main/java/teammates/ui/automated/AutomatedAction.java

```diff
@@ -58,10 +59,36 @@ protected String getRequestParamValue(String paramName) {
         return HttpRequestHelper.getValueFromRequestParameterMap(request, paramName);
     }
     
+    /**
+     * Retrieves the value for the specified parameter expected to be present in the http request.
+     *
+     * @param paramName  a constant from the {@link Const.ParamsNames} class.
+     * @return the value of the specified parameter.
+     * @throws NullPostParameterException if the parameter is not present in the http request.
```

Comment:
As this is an unchecked exception, no need to document. Instead, document the assumption i.e. 
`* Assumption: the requested parameter is not null`

The caller should know that violating assumptions results in an unchecked exception. Documenting the exact type of the exception is not needed because an assumption failure is considered a bug (this is like `assert` but GAE does not support asserts so we use our own mechanism) which means the dev needs to debug the code anyway.